tests/plugins/cloudservices/tokenregeneration.js

```diff
@@ -0,0 +1,71 @@
+/* bender-ckeditor-plugins: cloudservices */
+/* bender-include: %BASE_PATH%/plugins/easyimage/_helpers/tools.js*/
+/* global easyImageTools */
+
+( function() {
+	var TOKEN_VALUE = 'sample-token-value',
+		TOKEN_URL = '/mock_token_url',
+		MOCK_UPLOAD_URL = '/mock_upload_url',
+		xhrServer = sinon.fakeServer.create(),
+		counter = 0;
+
+	bender.editor = {
+		config: {
+			cloudServices_uploadUrl: MOCK_UPLOAD_URL,
+			cloudServices_tokenUrl: TOKEN_URL
+		}
+	};
+
+	CKEDITOR.once( 'instanceCreated', function( evt ) {
+		// Lower the polling rate.
+		evt.editor.CLOUD_SERVICES_TOKEN_INTERVAL = 100;
+	} );
+
+	xhrServer.respondWith( function( req ) {
```

Comment:
Code for creating fake token endpoint is present in several tests and it tends to be different for every test, yet – it always does the same. I propose creating generic `createFakeTokenEndpoint` function and extracting it to tests' helpers.

Reply:
I have intentionally not do that, as it would create another layer of abstraction on top of sinon that would be used in that way only in a single place, making it harder to understand for dev reading the code.

However given the amount of work I had to put into tests now, I decided to change the approach  in test suite, thus merging three test suites back into one.